Add installer, docs, hashbangs

pgunn opened the pull request.

pgunn commented:
I signed it!

mjanusz commented on setup.py:
```diff
@@ -0,0 +1,17 @@
+	scripts		= ['build_coordinates.py', 'compute_partitions.py', 'run_inference.py', 'train.py'],
+	url		= 'https://github.com/google/ffn',
+	license		= 'LICENSE',
+	description	= 'Convolutional Flood-Fill Networks',
```
Could you please change this to "Flood-Filling Networks for volumetric instance segmentation"?

mjanusz commented on setup.py:
```diff
@@ -0,0 +1,17 @@
+
+setup(
+	name		= 'ffn',
+	version		= '0.1.0',
```
Does this have to be specified? We are not planning on doing releases with this type of versioning scheme.

mjanusz commented on doc/inference_parameters.md:
```diff
@@ -0,0 +1,25 @@
+# ffn inference parameters
```
ffn -> FFN everywhere

mjanusz commented on doc/inference_parameters.md:
```diff
@@ -0,0 +1,25 @@
+normally invoked in a way that reads in such a file from the commandline. These are the
+parameters and what they mean:
+
+* image - A hash, usually with a single k-v pair of the string 'hdf5' to a path to an hdf5 file and dataset. Holds the raw image data against which inference is to be done
```
This is not a hash. `hdf5` is a field in `image`.

mjanusz commented on doc/inference_parameters.md:
```diff
@@ -0,0 +1,25 @@
+* image_stddev - In theory, the standard deviation of values in the targeted region. In practice, leave this at 33
+* checkpoint_interval - How often inference should dump its internal state to disk. If inference is interrupted, these checkpoints allow for interrupted work to be picked back up with the same parameters. In some environments this is more important so a lower value is warranted
+* seed_policy - Leave as "PolicyPeaks"
+* model_checkpoint_path - The path to the files that comprise the model used. Omit the final extension; if there are many versions of the model in the same directory you will probably want to pick the highest (most trained) number.
```
It's actually not completely correct that the highest numbered checkpoint is always the best. I believe we briefly mentioned this in the paper, but it probably deserves a more detailed discussion in a separate doc.

pgunn commented on doc/sample_workflow.md:
```diff
@@ -0,0 +1,34 @@
+# Setup and Process
+* The area of the global coordinate system that a segmentation dataset covers should be prepared into two hdf5 volumes, one with the raw data and one with the segmentation data. The segmentation data should be in an int64 collection in the volume. The raw data is often in a uint8 collection.
+* Run compute_partitions.py and build_coordinates as described in the readme; these transform the ground truth into a tensorflow record file (used to launch the network for training)
+* Run train.py, giving it the record file under train_coords, as well as the raw image data and the label volume. You should not need to adjust other parameters. This code continually drops new numbered network snapshots; if interrupted it can be restarted. It is best to leave it training for a very long time (even with high-end GPUs, it can be reasonable to leave it running for weeks or more dep ...
```
Ok, will revise.

mjanusz commented on doc/sample_workflow.md:
```diff
@@ -0,0 +1,34 @@
+* When you have that, do run_inference.py, passing it the protobuf-text config you just generated (as in the readme), as well as a bounding box spec (leave the start bit at zero, set the size to be the size of your inference target along each dimension). This will produce results in the directory you specified. It can take awhile
+* When the results are done, you will want to convert them into a format suitable for modification with other software (hdf5, or an image stack). A code snippet to do this is provided below.
+
+# Exporting ffn results
```
Perhaps it would be best to make this a standalone script, something like inference_to_hdf5.py?

mjanusz commented:
Thanks! I added a few more comments regarding the docs.

pgunn commented:
Sounds good. Even after addressing the review, I'll probably want to give the entire diff a good look over.

I expect to have the revisions in on Tuesday or Wednesday.

pgunn commented:
Addressed review, flattened diffs. Hoping we're good to go.
